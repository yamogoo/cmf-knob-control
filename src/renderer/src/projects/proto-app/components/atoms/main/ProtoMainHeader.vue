<template lang="pug">
header(
  :class="[`${PROTO_APP_PREFIX}-main-header`]"
)
  ProtoMainNavigation(
    data-testid="main-navigation"
  )
</template>

<script setup lang="ts">
import { PROTO_APP_PREFIX } from '@proto/config'

import ProtoMainNavigation from '@proto/components/atoms/main/ProtoMainNavigation.vue'
</script>

<style lang="scss">
$main-header: (
  sm: (
    size: (
      height: 48px
    )
  ),
  md: (
    size: (
      height: 60px
    )
  ),
  lg: (
    size: (
      height: 60px
    )
  )
);

.#{$PROTO_APP_PREFIX}-main-header {
  position: relative;
  display: block;
  width: 100%;
  background-color: none;
  overflow: hidden;

  @include respond-above(lg) {
    min-height: get('lg.size.height', $main-header);
  }

  @include respond-between(md, lg) {
    min-height: get('md.size.height', $main-header);
  }

  @include respond-below(md) {
    min-height: get('sm.size.height', $main-header);
  }
}
</style>
