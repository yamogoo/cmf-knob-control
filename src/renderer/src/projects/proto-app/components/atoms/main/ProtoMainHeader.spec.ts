import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'

import ProtoMainHeader from './ProtoMainHeader.vue'

describe('ProtoMainHeader', () => {
  test('header should render MainNavigation component', () => {
    const component = mount(ProtoMainHeader)

    const navEl = component.find('[data-test-id="main-navigation"]')
    const isNavElExist = navEl.exists()

    expect(isNavElExist).toBe(true)
    expect(isNavElExist).toMatchSnapshot()
  })
})
