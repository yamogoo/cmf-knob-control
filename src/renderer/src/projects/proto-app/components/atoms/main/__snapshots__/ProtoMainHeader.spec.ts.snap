// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProtoMainHeader > header should render MainNavigation component 1`] = `true`;
